src/storage/single_file_block_manager.cpp

```diff
@@ -282,6 +282,12 @@ void SingleFileBlockManager::Initialize(DatabaseHeader &header) {
 	meta_block = header.meta_block;
 	iteration_count = header.iteration;
 	max_block = header.block_count;
+
+	if (GetBlockAllocSize() != header.block_alloc_size) {
```

Comment:
This should probably only fail if the block size is set explicitly by the user, e.g. this should fail:
```sql
ATTACH 'dbname.db' (BLOCK_SIZE 16000);
DETACH dbname;
ATTACH 'dbname.db' (BLOCK_SIZE 32000);
```

But this should work:

```sql
ATTACH 'dbname.db' (BLOCK_SIZE 16000);
DETACH dbname;
ATTACH 'dbname.db';
```